mempool/cat/reactor.go

```diff
@@ -312,6 +315,36 @@ func (memR *Reactor) Receive(e p2p.Envelope) {
 			return
 		}
 
+		// Use local sequence information when available, otherwise query the application.
+		if len(msg.Signer) > 0 && msg.Sequence > 0 {
+			expectedSeq, haveExpected := memR.mempool.lowestSequenceForSigner(msg.Signer)
+			sequenceSource := "local-mempool"
+
+			if !haveExpected {
+				ctx := context.Background()
+				resp, err := memR.mempool.proxyAppConn.QuerySequence(ctx, &abci.RequestQuerySequence{
+					Signer: msg.Signer,
+				})
+				if err == nil && resp != nil && resp.Sequence > 0 {
```

Comment:
Is the first sequence 0?

Reply:
hmm interesting I saw 1 in the tests, but zero does make more sense

maybe that's just due to some initialization logic. I'll take a deeper dive

Reply:
updated to > https://github.com/celestiaorg/celestia-core/pull/2569/commits/f36f96f3598674e75ef0a2b1ee3d42138e7d36d2

the branch upstream already had this fixed as well.